job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContext.groovy

```diff
@@ -944,17 +944,17 @@ class PublisherContext extends AbstractExtensibleContext {
      *
      * @since 1.23
      */
-    @RequiresPlugin(id = 'stashNotifier')
+    @RequiresPlugin(id = 'stashNotifier', minimumVersion = '1.8')
     void stashNotifier(@DslContext(StashNotifierContext) Closure stashNotifierClosure = null) {
         StashNotifierContext context = new StashNotifierContext()
         ContextHelper.executeInContext(stashNotifierClosure, context)
 
         publisherNodes << new NodeBuilder().'org.jenkinsci.plugins.stashNotifier.StashNotifier' {
-            stashServerBaseUrl()
-            stashUserName()
-            stashUserPassword()
-            ignoreUnverifiedSSLPeer(false)
+            stashServerBaseUrl(context.stashServerBaseUrl)
+            stashUserName(context.stashUserName)
+            stashUserPassword(context.stashUserPassword)
             commitSha1(context.commitSha1)
+            ignoreUnverifiedSSLPeer(context.ignoreUnverifiedSSLPeer)
```

Comment:
New elements must only be generated when a matching plugin version is installed, see https://github.com/jenkinsci/job-dsl-plugin/blob/job-dsl-1.37/job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContext.groovy#L316